plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/resource/ResourcePayloadPage.java

```diff
@@ -0,0 +1,216 @@
+import org.eclipse.core.databinding.beans.BeanProperties;
+import org.eclipse.core.databinding.observable.value.IObservableValue;
+import org.eclipse.core.databinding.validation.IValidator;
+import org.eclipse.core.databinding.validation.MultiValidator;
+import org.eclipse.core.databinding.validation.ValidationStatus;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.jface.databinding.fieldassist.ControlDecorationSupport;
+import org.eclipse.jface.databinding.swt.WidgetProperties;
+import org.eclipse.jface.dialogs.ErrorDialog;
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.jface.layout.GridLayoutFactory;
+import org.eclipse.jface.wizard.IWizard;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.events.SelectionListener;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.FileDialog;
+import org.eclipse.swt.widgets.Group;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Text;
+import org.eclipse.ui.dialogs.ElementTreeSelectionDialog;
+import org.jboss.tools.common.ui.databinding.ParametrizableWizardPageSupport;
+import org.jboss.tools.common.ui.databinding.SimpleUrlStringValidator;
+import org.jboss.tools.common.ui.databinding.ValueBindingBuilder;
+import org.jboss.tools.openshift.common.core.utils.StringUtils;
+import org.jboss.tools.openshift.common.core.utils.VariablesHelper;
+import org.jboss.tools.openshift.internal.common.ui.databinding.RequiredControlDecorationUpdater;
+import org.jboss.tools.openshift.internal.common.ui.utils.FileValidator;
+import org.jboss.tools.openshift.internal.common.ui.utils.UIUtils;
+import org.jboss.tools.openshift.internal.ui.OpenShiftUIActivator;
+import org.jboss.tools.openshift.internal.ui.wizard.common.AbstractProjectPage;
+
+import com.openshift.restclient.OpenShiftException;
+
+/**
+ * @author Jeff Maury
+ *
+ */
+public class ResourcePayloadPage extends AbstractProjectPage<IResourcePayloadPageModel> {
+    
+    private static final IValidator URL_VALIDATOR =  new SimpleUrlStringValidator();
+
+    public ResourcePayloadPage(IWizard wizard, IResourcePayloadPageModel model) {
+        super(wizard, model,
+              "Select resource payload",
+              "Select the file or workspace file from where the resource will be created",
+              "resourcePayload");
+    }
+
+    @Override
+    protected void doCreateControls(Composite parent, DataBindingContext dbc) {
+        super.doCreateControls(parent, dbc);
+        createPayloadSourceControls(parent, dbc);
+    }
+
+    private void createPayloadSourceControls(Composite parent, DataBindingContext dbc) {
+        Group sourceGroup = new Group(parent, SWT.NONE);
```

Comment:
Unfortunately you hit the know issue with decorations in groups: an erroneous 2nd decoration is painted and updates dont work at time: https://bugs.eclipse.org/bugs/show_bug.cgi?id=478618
![image](https://cloud.githubusercontent.com/assets/25126/15015277/325f0280-120b-11e6-84c6-3c2c1a0bff34.png)
The bug was fixed (they say) in M7 which according to @mickaelistria will make it to our target platform. I'd therefore opt to NOT work around it as we usually do :)
Commenting here so that you know about it if QE complains about it.